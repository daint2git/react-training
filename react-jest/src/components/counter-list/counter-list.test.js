import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import CounterList from './counter-list'

describe('CounterList', () => {
  test('matches the snapshot', () => {
    const tree = renderer.create(<CounterList />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('shallow vs mount', () => {
    const shallowWrapper = shallow(<CounterList />)
    const mountWrapper = mount(<CounterList />)

    console.log('shallow html', shallowWrapper.debug())
    console.log('mount html', mountWrapper.debug())
  })

  test('should render two counters by default', () => {
    const wrapper = shallow(<CounterList />)
    const counters = wrapper.find('Counter')
    expect(counters.length).toEqual(2)
  })

  test('add more counters when user click the button', () => {
    const wrapper = shallow(<CounterList />)
    const addBtn = wrapper.find('Button')
    addBtn.simulate('click')
    const counters = wrapper.find('Counter')
    expect(counters.length).toEqual(3)
  })
})
