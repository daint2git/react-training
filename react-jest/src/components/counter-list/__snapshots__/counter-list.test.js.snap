// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterList matches the snapshot 1`] = `
Array [
  <ul>
    <li>
      <div
        className="counter"
      >
        <p>
          Current count: 
          0
        </p>
        <button
          name="increment"
          onClick={[Function]}
        >
          increment
        </button>
        <button
          name="decrement"
          onClick={[Function]}
        >
          decrement
        </button>
      </div>
    </li>
    <li>
      <div
        className="counter"
      >
        <p>
          Current count: 
          0
        </p>
        <button
          name="increment"
          onClick={[Function]}
        >
          increment
        </button>
        <button
          name="decrement"
          onClick={[Function]}
        >
          decrement
        </button>
      </div>
    </li>
  </ul>,
  <button
    onClick={[Function]}
    type="button"
  >
    add Counter
  </button>,
]
`;
